where-to-travel/src/main/webapp/script.js

```diff
@@ -1,212 +1,401 @@
-    "stylers": [
+    featureType: 'transit',
+    stylers: [
       {
-        "visibility": "off"
+        visibility: 'off'
       }
     ]
   }
 ];
-//End map stylings
+// End map stylings
 
-const submitId = "submit-id";
-const hoursId = "hrs";
-const minutesId = "mnts";
+// Thresholds for termination of search algorithm
+const placesThreshold = 30;
+const attemptsThreshold = 5;
+const directionThreshold = 5;
+
+// Document ids for user input elements
+const submitId = 'submit';
+const hoursId = 'hrs';
+const minutesId = 'mnts';
 
 let map;
 let home = null;
+let homeMarker = null;
+let markers = [];
 
 // Add gmap js library to head of page
-let script = document.createElement('script');
-script.src = 'https://maps.googleapis.com/maps/api/js?key=' 
-              + secrets['googleMapsKey'] + '&libraries=places';
+const script = document.createElement('script');
+script.src =
+  'https://maps.googleapis.com/maps/api/js?key=' +
+  secrets['googleMapsKey'] +
+  '&libraries=places';
 script.defer = true;
 script.async = true;
 
 document.head.appendChild(script);
 
-/**
- * Initializes map window, runs on load.
- */
+/** Initializes map window, runs on load. */
 async function initialize() {
-  let submit = document.getElementById(submitId);
+  const submit = document.getElementById(submitId);
   submit.addEventListener('click', submitDataListener);
-  home = await getUserLocation();    
-
+ 
   const mapOptions = {
-    center: home, 
+    center: {lat: 36.150813, lng: -40.352239},
     mapTypeId: google.maps.MapTypeId.ROADMAP,
-    zoom: 16,
+    zoom: 4,
     mapTypeControl: false,
-    styles: mapStyles
-  };	 
-  map = new google.maps.Map(document.getElementById("map"), mapOptions);
-
-  let homeMarker = new google.maps.Marker({
-    position: home,
-    map: map,
-    title: 'Home'
-  });
+    styles: mapStyles,
+  };
+
+  map = new google.maps.Map(document.getElementById('map'), mapOptions); 
+
+  showInfoModal();
 }
 
-/**
- * Responds to click on submit button by getting places and placing pins.
- *
- * @param {Event} event click event to respond to
+/** 
+ * Populates and opens modal with html content from info.txt that provides
+ * description of website to user
  */
-function submitDataListener(event) {
-  const hours = document.getElementById(hoursId).value;
-  const minutes = document.getElementById(minutesId).value;
-  const timeObj = { "hours": hours, "minutes": minutes };
-  getPlacesFromTime(timeObj).then(places => populatePlaces(places));
+function showInfoModal() {
+  fetch('info.txt')
+    .then(response => response.text())
+    .then(content => openModal(content));
 }
 
 /**
- * Populates map with pins. Given a list of places, put markers at each
- * lat/long location.
- *
- * @param {array} placeArray array of places to place markers at
+ * Obtains user's location from browser and sets home location. If error
+ * occurs, message is displayed to user in modal.
  */
-function populatePlaces(placeArray) {
```

Comment:
We will still need this method, but I add it back in my PR. this should be fine, though we will need to make sure we merge the conflict properly.

Reply:
Nevermind, it is defined later.